tpr/kind.go

```diff
@@ -0,0 +1,40 @@
+package tpr
+
+import (
+	"strings"
+
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+/*
+	Code based on:
+	https://github.com/kubernetes/kubernetes/blob/6a4d5cd7cc58e28c20ca133dab7b0e9e56192fe3/pkg/registry/extensions/thirdpartyresourcedata/util.go
+*/
+
+// extractKindAndGroup extracts kind and group from a name. For details see
+// Config.Name godoc.
+func extractKindAndGroup(name string) (kind string, group string, err error) {
```

Comment:
What do you think about doing `(kind, group string, err error)`?

/bikeshedding